### PR TITLE
Add transfer contract docs/spec

### PR DESCRIPTION
**Pull request by ureeves**
These docs will be added to the wiki in the end, so pay no attention to commit aesthetics. This will never be merged. I'm open to any and all feedback on this doc, be in in form or in content.

The easiest way of actually seeing what it looks like is by going to the [file in the `transfer-docs` branch].

[file in the `transfer-docs` branch]: https://github.com/dusk-network/rusk/blob/transfer-docs/contracts/docs/transfer.md

Resolves: #2058

### CODE REVIEW

**Review by HDauven — DISMISSED** [review]

File: contracts/docs/transfer.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
+# Transfer Contract
+
+## Introduction
+
+The transfer contract is one of Dusk's genesis smart contracts, and is responsible for defining the
+rules that a transaction must follow to be executed on the network. It serves as an entrypoint for
+every transaction, and can therefore be though as the definition of 'Dusk' as a cryptocurrency.
```
Comment (HDauven) [inline contracts/docs/transfer.md:7]: `though`-> `thought of`

File: contracts/docs/transfer.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
+# Transfer Contract
+
+## Introduction
+
+The transfer contract is one of Dusk's genesis smart contracts, and is responsible for defining the
+rules that a transaction must follow to be executed on the network. It serves as an entrypoint for
```
[the remark this answers is not in the document]
Comment (ureeves) [reply contracts/docs/transfer.md:6]: It's a style thing. Makes reading easier IMO, but it's subjective.

File: contracts/docs/transfer.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+every transaction, and can therefore be though as the definition of 'Dusk' as a cryptocurrency.
+
+Like any smart contract on Dusk, the transfer contract exposes a set of functions and maintains
+state. In this document we will go through the functions it exposes, and discuss their intended
```
[the remark this answers is not in the document]
Comment (ureeves) [reply contracts/docs/transfer.md:10]: Yes

File: contracts/docs/transfer.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+the functions operate on, and describe the types of mutation to the state each of them may perform.
+
+## State
+
+The state of the transfer contract consists of the state necessary to ensure the security of each
+transaction, as well as provide data for wallets wishing to create one.
+
+- Merkle tree of all Phoenix notes (UTXO)
+- Set of nullifiers
+- Roots of the merkle tree
```
[the remark this answers is not in the document]
Comment (ureeves) [reply contracts/docs/transfer.md:21]: Fixed for all instances

File: contracts/docs/transfer.md [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+every transaction, and can therefore be though as the definition of 'Dusk' as a cryptocurrency.
+
+Like any smart contract on Dusk, the transfer contract exposes a set of functions and maintains
+state. In this document we will go through the functions it exposes, and discuss their intended
+use, together with their arguments, returns, and calling convention. We will also specify the state
+the functions operate on, and describe the types of mutation to the state each of them may perform.
+
+## State
+
+The state of the transfer contract consists of the state necessary to ensure the security of each
```
[the remark this answers is not in the document]
Comment (ureeves) [reply contracts/docs/transfer.md:16]: I'll phrase it differently to avoid repetition of the word "state"

File: contracts/docs/transfer.md (hunk 1 above)
Comment (ureeves) [reply contracts/docs/transfer.md:7]: 👍

**Pull request closed without merging**